ui/panels-plugin/src/plugins/time-series-table/DataTable.tsx

```diff
@@ -0,0 +1,131 @@
+// limitations under the License.
+/*
+ eslint-disable @typescript-eslint/no-explicit-any
+ */
+import { Fragment, ReactNode } from 'react';
+import { Alert, Table } from '@mui/material';
+import { TimeSeriesData } from '@perses-dev/core';
+import { QueryData } from '@perses-dev/plugin-system';
+
+export interface DataTableProps {
+  result: Array<QueryData<TimeSeriesData>>;
+}
+
+/**
+ * Designed to display timeseries data in a prometheus like table format.
+ * The first column will contain the metric name and label combination, and the second column will contain the values.
+ * This is inspired by prometheus DataTable.
+ * https://github.com/prometheus/prometheus/blob/2524a915915d7eb1b1207152d2e0ce5771193404/web/ui/react-app/src/pages/graph/DataTable.tsx
+ * @param result timeseries query result
+ * @constructor
+ */
+const DataTable = ({ result }: DataTableProps) => {
+  if (!result) {
+    return null;
+  }
+  const series = result.flatMap((d) => d.data).flatMap((d) => d?.series || []);
+  const maxFormattableSize = 1000;
+  let rows: ReactNode[] = [];
+  let limited = false;
+  const doFormat = series.length <= maxFormattableSize;
+  rows = series.map((s, seriesIdx) => {
+    const displayTimeStamps = (s.values?.length || 0) > 1;
+    const valuesAndTimes = s.values
+      ? s.values.map((v, valIdx) => {
+          return (
+            <Fragment key={valIdx}>
+              {v[1]} {displayTimeStamps && <span>@{v[0]}</span>}
+              <br />
+            </Fragment>
+          );
+        })
+      : [];
+    const histogramsAndTimes = s.histograms
+      ? s.histograms.map((h: any, hisIdx: number) => {
+          return (
+            <Fragment key={-hisIdx}>
+              {histogramTable(h[1])} @<span>{h[0]}</span>
+              <br />
+            </Fragment>
+          );
+        })
+      : [];
+    return (
+      <tr style={{ whiteSpace: 'pre' }} key={seriesIdx}>
+        <td>{s.formattedName || s.name}</td>
+        <td>
+          {valuesAndTimes} {histogramsAndTimes}
+        </td>
+      </tr>
+    );
+  });
+  limited = rows.length !== series.length;
+
+  return (
+    <>
+      {limited && (
+        <Alert>
+          <strong>Warning:</strong> Fetched {series.length} metrics, only displaying first {rows.length}.
+        </Alert>
+      )}
+      {!doFormat && (
+        <Alert>
+          <strong>Notice:</strong> Showing more than {maxFormattableSize} series, turning off label formatting for
+          performance reasons.
+        </Alert>
+      )}
+      <Table className="data-table">
+        <tbody>{rows}</tbody>
+      </Table>
+    </>
+  );
+};
+
+const leftDelim = (br: number): string => (br === 3 || br === 1 ? '[' : '(');
+const rightDelim = (br: number): string => (br === 3 || br === 0 ? ']' : ')');
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const bucketRangeString = ([boundaryRule, leftBoundary, rightBoundary, _]: [
```

Comment:
Because of the "\_" I guess we need to tweak the rule to skip check on "\_" vars

Reply:
Updated the eslint rules